### ISSUE TITLE
E3724 rejects an intrinsic in Globals.Function.PermissionsBoundary that CloudFormation and the resource-level property accept

### ISSUE THREAD

**Issue by linsoraxpress** [aws-cloudformation/cfn-lint#4680]
### CloudFormation Lint Version

cfn-lint 1.56.0 (also 1.54.0 and 1.55.1, i.e. every release since #4491; reproduced on `main` at f4c72e0cf7). cfn-lint 1.53.3 (the last transform-based release) passes the same template.

### What operating system are you using?

macOS 15 (Darwin 25.6), Python 3.14 via `uvx cfn-lint@1.56.0`; also reproduced on Ubuntu 24.04.

### Describe the bug

Since #4491 the `Globals` section is validated against the bundled SAM Globals schema (`src/cfnlint/data/schemas/other/sam/globals.json`). In that schema `properties.Function.properties.PermissionsBoundary` is `{"type": "string"}`, while the sibling Globals properties that also take strings or ARNs (`Layers`, `Environment`, `Timeout`, ...) are `{}`. As a result E3724 rejects any intrinsic function in `Globals.Function.PermissionsBoundary`:

```text
E3724 {'Fn::ImportValue': {'Fn::Sub': 'my-iam-${Environment}-lambda-boundary-arn'}} is not of type 'string'
template.yaml:9:5
```

CloudFormation accepts this template: the SAM transform merges the Globals value into every `AWS::Serverless::Function`, whose `PermissionsBoundary` property takes an intrinsic (the SAM Globals reference lists `PermissionsBoundary` as a supported Globals property, and `sam validate`, which runs the real translator, passes the template). The same `!ImportValue` declared directly on the function instead of in `Globals` passes cfn-lint 1.56.0 with no finding, so only the Globals path is affected. Importing the boundary ARN from a separate IAM stack is the usual shape for a fleet that shares one permissions boundary (every Lambda stack in our fleet does it), so every one of those templates now fails E3724 on a non-defect.

Reproduction commands and results:

```text
$ uvx cfn-lint@1.56.0 --regions us-east-1 -t template.yaml
E3724 {'Fn::ImportValue': {'Fn::Sub': 'my-iam-${Environment}-lambda-boundary-arn'}} is not of type 'string'
template.yaml:9:5
(exit 2)

$ uvx --from git+https://github.com/aws-cloudformation/cfn-lint cfn-lint --regions us-east-1 -t template.yaml   # main @ f4c72e0cf7
E3724 ... is not of type 'string'
(exit 2)

$ uvx cfn-lint@1.53.3 --regions us-east-1 -t template.yaml
(exit 0)

$ sam validate --region us-east-1     # SAM CLI 1.165.0, aws-sam-translator 1.111.0
template.yaml is a valid SAM Template.
```

Moving the same `PermissionsBoundary: !ImportValue ...` from `Globals.Function` onto the function's own `Properties` makes cfn-lint 1.56.0 exit 0.

### Expected behavior

No finding: `Globals.Function.PermissionsBoundary` accepts the same values as `AWS::Serverless::Function.PermissionsBoundary`, including `Fn::ImportValue`, `Fn::Sub`, and `Ref`. A fix could type the property like its intrinsic-capable siblings in `globals.json` (or validate Globals values through the same intrinsic-aware path the resource properties use), so an intrinsic in a string-valued Globals property is validated rather than rejected outright.

### Reproduction template

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31
Parameters:
  Environment:
    Type: String
Globals:
  Function:
    Runtime: nodejs22.x
    PermissionsBoundary: !ImportValue
      Fn::Sub: my-iam-${Environment}-lambda-boundary-arn
Resources:
  HelloFunction:
    Type: AWS::Serverless::Function
    Properties:
      Handler: index.handler
      InlineCode: "exports.handler = async () => ({ statusCode: 200 });"
```
